### PR TITLE
Wait for activation code to complete

### PR DESCRIPTION
**Pull request by DonJayamanne**
Part of #4050
Wait for `activate` stuff to complete before returning.

### CODE REVIEW

File: src/client/extension.ts [hunk 1]
```diff
@@ -65,6 +65,7 @@ export async function activate(context: IExtensionContext): Promise<IExtensionAp
     sendStartupTelemetry(ready, durations, stopWatch, serviceContainer)
         // Run in the background.
         .ignoreErrors();
+    await ready;
```
Comment (DonJayamanne) [inline src/client/extension.ts:68]: This is what is crucial

File: src/client/datascience/notebook/integration.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export class NotebookIntegration implements IExtensionSingleActivationService {
         // This condition is temporary.
         // If user belongs to the experiment, then make the necessary changes to package.json.
         // Once the API is final, we won't need to modify the package.json.
-        if (await this.experimentService.inExperiment(Experiments.NativeNotebook)) {
+        if (this.useNativeNb) {
```
Comment (DonJayamanne) [inline src/client/datascience/notebook/integration.ts:47]: While I was at it I used existing value instead of checking experiments.

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

File: src/client/migration/migrateDigestStorage.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export class MigrateDigestStorage {
             try {
                 await this.ensuredDir;
                 const nbsecret = path.join(this.pythonExtensionStorageDir, 'nbsecret');
-                if (this.fs.localDirectoryExists(nbsecret)) {
+                if (await this.fs.localDirectoryExists(nbsecret)) {
```
Comment (DonJayamanne) [inline src/client/migration/migrateDigestStorage.ts:32]: Another bug fix

File: src/client/migration/migrateDigestStorage.ts [hunk 4]
```diff
@@ -48,7 +48,7 @@ export class MigrateDigestStorage {
             try {
                 await this.ensuredDir;
                 const nbsignatures = path.join(this.pythonExtensionStorageDir, 'nbsignatures');
-                if (this.fs.localDirectoryExists(nbsignatures)) {
+                if (await this.fs.localDirectoryExists(nbsignatures)) {
```
Comment (DonJayamanne) [inline src/client/migration/migrateDigestStorage.ts:51]: Another bug fix

Comment (joyceerhl) [reply src/client/migration/migrateDigestStorage.ts:51]: Yikes, thanks for fixing this

**Pull request merged**